contrib/babelfishpg_common/src/varchar.c

```diff
@@ -1102,6 +1110,82 @@ varchar2numeric(PG_FUNCTION_ARGS)
 	PG_RETURN_NUMERIC(result);
 }
 
+#define FIXEDDECIMAL_MULTIPLIER_NEW 100LL
+#define FIXEDDECIMAL_SCALE_NEW 2
+Datum
+fixeddecimal2varchar(PG_FUNCTION_ARGS)
+{
+	int64		val = PG_GETARG_INT64(0);
+	int32		maxByteLen = PG_GETARG_INT32(1);
+	char		buf[MAXINT8LEN + 1];
+	char	   *end;
+	int32		len;
+	Datum		res;
+
+	/* fetch function pointer for cross-module calls. */
+	if (fixeddecimal2str_p == NULL)
+		fixeddecimal2str_p = (fixeddecimal2str_t)
+			load_external_function("$libdir/babelfishpg_money", "fixeddecimal2str", true, NULL);
+
+	end = fixeddecimal2str_p(val, buf, FIXEDDECIMAL_MULTIPLIER_NEW, FIXEDDECIMAL_SCALE_NEW);
+	len = (end - buf);
+
+	maxByteLen -= VARHDRSZ;
+	if (maxByteLen >= 0 && len > maxByteLen)
+		ereport(ERROR,
+				(errcode(ERRCODE_STRING_DATA_RIGHT_TRUNCATION),
+				 errmsg("There is insufficient result space to convert a money/smallmoney value to varchar.")));
```

Comment:
to varchar/nvarchar

Reply:
Anyway to test this condition?

Reply:
1. This is always `varchar` in T-SQL for all string types.
2. There is already one test case in test_conv_money_to_varchar-vu-verify file.